### PR TITLE
Added assets bulk delete api

### PR DESCRIPTION
**Pull request by tushar-balwani**
# Description

Added **`bulk delete`** api in TIO assets
```
assets_info = tio.assets.bulk_delete(('host.hostname', 'match', 'asset.com'), filter_type='or')
```

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- Added below tests for assets bulk delete api
- [x] `test_assets_bulk_delete_filter_type_typeerror`
- [x] `test_assets_bulk_delete_filter_type_unexpectedvalueerror`
- [x] `test_assets_bulk_delete_bad_filter`
- [x] `test_assets_bulk_delete_success`

**Test Configuration**:
* Python Version(s) Tested: 3.8.6
* Tenable.sc version (if necessary):

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: tenable/io/base.py [hunk 1]
```diff
@@ -110,6 +128,9 @@ def _wait_for_download(self, path, resource, resource_id, file_id, **kw):
         return status
 
 class TIOIterator(APIResultsIterator):
+    '''
+    The following methods allows us to iterate through pages and get data
+    '''
```
Comment (mvalliammal) [inline tenable/io/base.py:133]: Where the self._query is declared?

File: tests/io/test_base.py [hunk 2]
```diff
@@ -1,39 +1,69 @@
...
+def test_base_parse_filter_assets(api, fitem, fset):
+    '''
+    test to parse filters with rtype as assets
+    '''
+    fitem = [(fitem[0][0], fitem[0][1], [fitem[0][2]])]
+    assert {'asset': [{
+        'field': 'distro',
+        'operator': 'match',
+        'value': 'win'
+    }]} == api.agents._parse_filters(fitem, fset, rtype='assets')
```
Comment (mvalliammal) [inline tests/io/test_base.py:69]: Access to protected member _parse_filters of a client class  agents

Comment (tushar-balwani) [reply tests/io/test_base.py]: Addressed above comment in next commit

**Review by valliammal — APPROVED** [review]
I approved the changes

**Review by levwais — APPROVED** [review]

**Pull request merged**